File: MuseoPOO/escultura.py
"""Módulo que define la clase Escultura."""

from datetime import date

from obra_de_arte import ObraDeArte


class Escultura(ObraDeArte):
    """Representa una escultura del museo."""

    def __init__(
        self,
        id_obra: int,
        titulo: str,
        autor: str,
        periodo: str,
        valoracion: float,
        fecha_creacion: date,
        fecha_entrada: date,
        estilo: str,
        material: str,
        estado: str = "expuesta",
    ) -> None:
        super().__init__(
            id_obra, titulo, autor, periodo,
            valoracion, fecha_creacion, fecha_entrada, estado,
        )
        self._estilo = estilo
        self._material = material

    def get_estilo(self) -> str:
        return self._estilo

    def get_material(self) -> str:
        return self._material

    def consultar_info(self) -> str:
        return (
            f"Escultura: '{self._titulo}' | Autor: {self._autor} | "
            f"Periodo: {self._periodo} | Estilo: {self._estilo} | "
            f"Material: {self._material} | "
            f"Valoración: ${self._valoracion:,.2f} | "
            f"Estado: {self._estado}"
        )

File: Museo Java/Escultura.java
import java.time.LocalDate;

public class Escultura extends ObraDeArte {

    private String estilo;
    private String material;

    public Escultura(int idObra, String titulo, String autor, String periodo,
                     double valoracion, LocalDate fechaCreacion, LocalDate fechaEntrada,
                     String estilo, String material, String estado) {
        super(idObra, titulo, autor, periodo, valoracion, fechaCreacion, fechaEntrada, estado);
        this.estilo = estilo;
        this.material = material;
    }

    public Escultura(int idObra, String titulo, String autor, String periodo,
                     double valoracion, LocalDate fechaCreacion, LocalDate fechaEntrada,
                     String estilo, String material) {
        this(idObra, titulo, autor, periodo, valoracion, fechaCreacion, fechaEntrada, estilo, material, "expuesta");
    }

    public String getEstilo() { return estilo; }
    public String getMaterial() { return material; }

    @Override
    public String consultarInfo() {
        return String.format("Escultura: '%s' | Autor: %s | Periodo: %s | Estilo: %s | Material: %s | Valoración: $%,.2f | Estado: %s",
                titulo, autor, periodo, estilo, material, valoracion, estado);
    }
}
